cql3/functions/aggregate_fcts.cc

```diff
@@ -123,6 +123,26 @@ wrap_function_autonull(null_handler nullhandler, Ret (*func)(Args...)) {
     };
 }
 
+template <typename Ret, typename... Args>
+shared_ptr<scalar_function>
+make_internal_scalar_function(sstring name, null_handler nullhandler, Ret (*func)(Args...)) {
+    return ::make_shared<internal_scalar_function>(
+            std::move(name),
+            data_type_for<Ret>(),
+            std::vector({data_type_for<Args>()...}),
+            wrap_function_autonull(nullhandler, func)
+    );
+}
+
+template <typename Lambda>
+requires std::is_class_v<Lambda>
+shared_ptr<scalar_function>
+make_internal_scalar_function(sstring name, null_handler nullhandler, Lambda func) {
+    // "+func" decays the lambda into a pointer-to-function, so that its signature
+    // can be inferred by the other overload.
```

Comment:
Wow, every time I think that maybe, just maybe, I'm done learning C++, you come up with yet another feature I never heard of ;-)

Reply:
It's not really a feature, just an unexpected side effect. Unary operator + is defined for pointers, but not for class types. So C++ tries to find an implicit conversion that makes the unary operator + work, finds the implicit conversion to function pointer, and applies it.